### ISSUE TITLE
No default export found in imported module "vike/plugin"

### ISSUE THREAD

**Issue by rtritto** [vikejs/vike#1838]
### Description

Using ESLint with `eslint-import-resolver-typescript` (under the hood _eslint-import-resolver-node_), the rule `import/default` of `eslint-import-resolver-node` fails with `vike/plugin` import.

## Code
- [.eslintrc.yml](https://github.com/rtritto/template-vike-solid-daisyui/blob/master/.eslintrc.yml)

- [tsconfig.json](https://github.com/rtritto/template-vike-solid-daisyui/blob/master/tsconfig.json)
  Note: _compilerOptions.esModuleInterop_ is _true_

- [vite.config.ts](https://github.com/rtritto/template-vike-solid-daisyui/blob/master/vite.config.ts)
  ```ts
  import vike from 'vike/plugin'
  ```

## Output
```console
1:8  error  No default export found in imported module "vike/plugin"  import/default
```

In VSCode, opening the reference of `vike/plugin` (/vike/dist/esm/node/plugin/index.d.ts), I got:

`vite.config.ts`
```ts
export default plugin;
export { plugin };
export { plugin as ssr };
export type { ConfigVikeUserProvided as UserConfig };
export { PROJECT_VERSION as version } from './utils.js';
import type { ConfigVikeUserProvided } from '../../shared/ConfigVike.js';
declare function plugin(vikeConfig?: ConfigVikeUserProvided): any;
```

## Reproduction
1. `git clone https://github.com/rtritto/template-vike-solid-daisyui.git`
2. `cd template-vike-solid-daisyui`
3. `yarn`
4. `yarn eslint .`

**Comment by rtritto** [comment]
Fixed with:
```ts
import { plugin } from 'vike/plugin'
```

**Comment by rtritto** [comment]
Maybe it can be fixed moving the `declare ...` before the `export default ...`

**Comment by brillout** [comment]
I cannot reproduce:

```
~/tmp/template-vike-solid-daisyui (master|u+1) pe eslint .

🌼   daisyUI 4.12.10
├─ ✔︎ 1 theme added		https://daisyui.com/docs/themes
╰─ ❤︎ Support daisyUI project:	https://opencollective.com/daisyui


🌼   daisyUI 4.12.10
├─ ✔︎ 1 theme added		https://daisyui.com/docs/themes
╰─ ★ Star daisyUI on GitHub	https://github.com/saadeghi/daisyui
```

Note that I'm using pnpm instead of yarn: https://github.com/brillout/template-vike-solid-daisyui. Can you reproduce with pnpm instead of yarn?

**Comment by rtritto** [comment]
> Fixed changing import from _default_ to _named_:
> 
> ```ts
> import { plugin as vike } from 'vike/plugin'
> ```
> 
> Same issue with `vike-solid/config` (`esModuleInterop: false`, missing default export) and `vike-node/plugin`
> 
> FYI @brillout

Sorry, on _main branch_, I changed the code with the naming import.
The issue will apper changing with the default import (you can try changing the code):
```diff
-import { plugin as vike } from 'vike/plugin'  // OK ESLint
+import vike from 'vike/plugin'  // KO ESLint
```

**Comment by brillout** [comment]
I could reproduce indeed and updated my fork.

Seems to be a ESLint quirk? Closing as I don't see this to be a Vike bug, but I'm open for a workaround.

**Comment by brillout** [comment]
Fix pre-released as `0.4.193-commit-c1d490a`.
